pkg/providers/ovirt/provider.go

```diff
@@ -225,7 +226,27 @@ func (o *OvirtProvider) FindTemplate() (*templatev1.Template, error) {
 
 // ProcessTemplate uses openshift api to process template
 func (o *OvirtProvider) ProcessTemplate(template *templatev1.Template, vmName *string, namespace string) (*kubevirtv1.VirtualMachine, error) {
-	return o.templateHandler.ProcessTemplate(template, vmName, namespace)
+	vm, err := o.templateHandler.ProcessTemplate(template, vmName, namespace)
+	if err != nil {
+		return nil, err
+	}
+	sourceVM, err := o.getVM()
+	if err != nil {
+		return nil, err
+	}
+	labels, err := o.templateFinder.GetLabels(sourceVM)
+	if err != nil {
+		return nil, err
+	}
+	return updateLabels(vm, labels, *vmName), nil
+}
+
+func updateLabels(vm *kubevirtv1.VirtualMachine, labels map[string]string, vmName string) *kubevirtv1.VirtualMachine {
+	utils.AddLabels(vm.ObjectMeta.GetLabels(), labels)
+
+	labels[vmNameLabel] = vmName
```

Comment:
this seems common to all providers - can we move it to https://github.com/kubevirt/vm-import-operator/pull/158/files#diff-c79b0981c3f458c6eaedf56f00a4e063R69 (pkg/templates/template-handler.go#addLabels )

Reply:
Update and the label list is common but the way how we obtain the information to build the label map is specific. I thought about it and failed to find good way to do it.

Reply:
I only referred to "vm.kubevirt.io/name" label.
There is a set of labels that we create in pkg/templates/template-handler.go#addLabels
"vm.kubevirt.io/name" requires the value of vmName which passed to o.templateHandler.ProcessTemplate(template, vmName, namespace) in line 229.

Reply:
OK, will move "vm.kubevirt.io/name" label.